integration-tests/equations/test_forced_advection.py

```diff
@@ -0,0 +1,88 @@
+"""
+Tests discretisations of the forced advection equation. This checks the
+errornorm for the resulting field against an analytic solution to ensure that
+the result is reasonable.
+"""
+
+from gusto import *
+from firedrake import (PeriodicIntervalMesh, SpatialCoordinate, FunctionSpace,
+                       VectorFunctionSpace, conditional, acos, cos, pi,
+                       as_vector, errornorm)
+
+
+def run_forced_advection(tmpdir):
+
+    # mesh, state and equation
+    Lx = 100
+    delta_x = 0.05
+    nx = int(Lx/delta_x)
+    mesh = PeriodicIntervalMesh(nx, Lx)
+    x = SpatialCoordinate(mesh)[0]
+
+    dt = 0.005
+    output = OutputParameters(dirname=str(tmpdir), dumpfreq=1)
+    diagnostic_fields = [CourantNumber()]
+
+    state = State(mesh,
+                  dt=dt,
+                  output=output,
+                  parameters=None,
+                  diagnostics=None,
+                  diagnostic_fields=diagnostic_fields)
+
+    VD = FunctionSpace(mesh, "DG", 1)
+    Vu = VectorFunctionSpace(mesh, "CG", 1)
+
+    # set up parameters and initial conditions
+    u_max = 1
+    C0 = 0.6
+    K0 = 0.3
+    Csat = 0.75
+    Ksat = 0.25
+    tmax = 85
+
+    # saturation profile
+    msat_expr = Csat + (Ksat * cos(2*pi*(x/Lx)))
+    msat = Function(VD)
+    msat.interpolate(msat_expr)
+
+    # initial moisture profile
+    mexpr = C0 + K0*cos((2*pi*x)/Lx)
+
+    rain = Rain(space='tracer',
+                transport_eqn=TransportEquationType.no_transport)
+    meqn = ForcedAdvectionEquation(state, VD, field_name="water_vapour", Vu=Vu,
+                                   active_tracers=[rain])
+    physics_schemes = [(InstantRain(meqn, msat, rain_name="rain",
+                                    set_tau_to_dt=True), ForwardEuler(state))]
+
+    state.fields("u").project(as_vector([u_max]))
+    qv = state.fields("water_vapour")
+    qv.project(mexpr)
+
+    # exact rainfall profile (analytically)
+    r_exact = state.fields("r_exact", VD)
+    lim1 = Lx/(2*pi) * acos((C0 + K0 - Csat)/Ksat)
+    lim2 = Lx/2
+    coord = (Ksat*cos(2*pi*x/Lx) + Csat - C0)/K0
+    exact_expr = 2*Ksat*sin(2*pi*x/Lx)*acos(coord)
+    r_expr = conditional(x < lim2, conditional(x > lim1, exact_expr, 0), 0)
+    r_exact.interpolate(r_expr)
+
+    # build time stepper
+    stepper = PrescribedTransport(meqn, RK4(state), state,
+                                  physics_schemes=physics_schemes)
+
+    stepper.run(0, tmax=tmax)
+
+    error = errornorm(r_exact, state.fields("rain"))
+
+    return error
+
+
+def test_forced_advection(tmpdir):
+
+    tol = 0.003
```

Comment:
With reduced resolution and increased time step, the test may no longer pass with this tolerance so we might need to change that too